controllers/AuthenticationController.ts

```diff
@@ -16,13 +16,16 @@ const AuthenticationController = (app: Express) => {
     const password = user.password;
     console.log(password);
     const existingUser = await userDao.findUserByUsername(username);
-    const match = await bcrypt.compare(password, existingUser.password);
-
-    if (match) {
-      existingUser.password = "*****";
-      // @ts-ignore
-      req.session["profile"] = existingUser;
-      res.json(existingUser);
+    if(existingUser) {
```

Comment:
Can you tell me why you moved the match condition inside the if condition?

Reply:
Added a check to display the error message related to the error occurring. If the user already exists in the DB, then instead of "Network Error", an error message related to this message will be thrown. 

Context: Previously the existingUser variable was used to store if the current user is already existing in the DB or not. But the variable was not used.